### ISSUE TITLE
Add react history v5 support

### ISSUE THREAD

**Issue by AlekseyManetov** [epam/UUI#817]
## Description
Add to UUI react history v5 support
## Describe the solution you'd like
UUI should work with react-router v6 and history v5 
https://github.com/remix-run/history
https://github.com/remix-run/react-router

**Comment by sliim35** [comment]
For whom it might need right away
I've created a little workaround with help of @jakobz,
Basically you need to implement `AdapterClass` on your own

```ts
class Adapter implements IRouterContext {
    // In order to use actions,loaders need to pass router object instead of just history
    constructor(private router: ReturnType<typeof createBrowserRouter>) {}

    public getCurrentLink(): Link {
        return Adapter.searchToQuery(this.router.state.location);
    }
    
    // Leave it as is
    public redirect(link: Link): void {}

    // history.push -> router.navigate
    public transfer(link: Link): void {
        this.router.navigate(HistoryV5AdaptedRouter.queryToSearch(link), {
            replace: true,
        });
    }
    
    // Leave it as is
    public isActive(link: Link): boolean {}

    // Leave it as is
    public createHref(): string {}

    // history.listen -> router.subscribe
    public listen(listener: (link: Link) => void) {
        return this.router.subscribe(({ location }) => listener(location));
    }

    // History v5 doesn't support yet block api
    // @link https://github.com/remix-run/react-router/issues/8139#issuecomment-1262630360
    public block(listener: (link: Link) => void) {}

    // Leave it as is
    private static searchToQuery(link: Link): Link {};
    }

    // Leave it as is
    private static queryToSearch(link: Link): Link {}
}
```
And use it something like that
```ts
const router = createBrowserRouter(
  createRoutesFromElements()
);

const uuiRouter = new HistoryAdaptedRouter(router);
const { services } = useUuiServices({
    apiDefinition: () => Promise.resolve({}),
    router: uuiRouter,
    skinContext,
});
```
Here is a sandbox [example](https://stackblitz.com/edit/vitejs-vite-zwnhha?file=src/App.tsx)

### Caveats
- Make sure that you turned off <React.StrictMode>
- Don't work block api and hence https://uui.epam.com/documents?id=lockContextDoc&mode=doc&skin=UUI4_promo&category=contexts
- you are using `react-router-dom` 6.4+
- you initialized your app using [hook version](https://uui.epam.com/documents?id=contextProvider&mode=doc&skin=UUI4_promo&category=contexts#example_with_hook)

**Comment by NatalliaAlieva** [comment]
Released in 5.1.1 ver.
